narwhal/primary/src/synchronizer.rs

```diff
@@ -67,6 +67,8 @@ struct Inner {
     payload_store: Store<(BatchDigest, WorkerId), PayloadToken>,
     /// Send missing certificates to the `CertificateFetcher`.
     tx_certificate_fetcher: Sender<Certificate>,
+    /// Send certificates to be accepted into a separate loop, to avoid client cancellations.
```

Comment:
I would suggest to add a few more info here on why it is important - meaning the potential atomicity risk where a certificate otherwise could get processed but not sent to consensus leading to an invalid state with missing parents. We need to make sure that anyone in the future touching this code clearly understand the subtle risks.

Reply:
Definitely, added more comments especially above `process_certificate_with_lock()`.